src/test/java/hw6/MetalAndColorsFormTest.java

```diff
@@ -0,0 +1,30 @@
+package hw6;
+
+
+import hw6.data.DataProviderTestData;
+import org.testng.annotations.Test;
+
+
+import static hw6.JdiSite.*;
+
+public class MetalAndColorsFormTest extends AbstractJdiTest {
+
+
+    @Test(dataProviderClass = DataProviderTestData.class,
+            dataProvider = "getMetalsAndColorsTestData")
+    public void idiMetalAndColorsFormTest(int[] summary, String[] elements,
```

Comment:
Take a look at Entity Driving testing and Login form.
You need to create Site.Page.Form::submit(Entity) <- this structure.

Reply:
Changed sctructure. Try to apply ED to the current implementation.